### ISSUE TITLE
Sporadic 408 errors

### ISSUE THREAD

**Issue by rh2048** [cshum/imagor#297]
Our production Imagor instance starts returning a 408 responses after a seemingly random amount of time. It returns the following body:

```
{"message":"timeout","status":408}
```

Here's what the response codes look like. The green line is 200 status code, orange is 408. Each time it turns to green, I restarted the instance.

<img width="1368" alt="image" src="https://user-images.githubusercontent.com/96033673/212235039-dcf92a2f-84f5-4970-bee8-4cf518f8044b.png">

All of our requests are with the following filter:

```
/unsafe/512x512/filters:strip_exif():strip_icc():format(jpeg):quality(80):background_color(2a2a2a)/${url}
````

We use the following configuration:

```
"IMAGOR_PROCESS_CONCURRENCY": "1",
"S3_RESULT_STORAGE_BUCKET": "xxxx",
"AWS_RESULT_STORAGE_SECRET_ACCESS_KEY": "xxxx",
"S3_RESULT_STORAGE_EXPIRATION": "48h",
"VIPS_CONCURRENCY": "1",
"S3_FORCE_PATH_STYLE": "1",
"AWS_RESULT_STORAGE_ACCESS_KEY_ID": "xxxx",
"S3_RESULT_STORAGE_ENDPOINT": "xxxx",
"AWS_RESULT_STORAGE_REGION": "xxxx",
"IMAGOR_UNSAFE": "1"
```

We used to be running v1.2.5 without issue. This started happening after upgrading to 1.3.5.

**Comment by cshum** [comment]
`"IMAGOR_PROCESS_CONCURRENCY": "1"` means the request will queue up when there is more than 1 image processed at a time. They will 408 timeout eventually when there is too many requests in the queue. 

What about setting `IMAGOR_PROCESS_CONCURRENCY` to a higher number, say to the number of CPUs?

**Comment by NathanFlurry** [comment]
I haven't had a chance to try this out yet, it's on my list. Imagor is currently running on one core, so I need to increase the concurrency. Can we reopen this issue? Thanks!

**Comment by rh2048** [comment]
Updated our Imagor instance to 1.3.7 and deployed with 2 CPU cores and 2 GB of RAM. It's now running with `VIPS_CONCURRENCY=2` and `IMAGOR_PROCESS_CONCURRENCY=2`.

Will let you know tomorrow if it's fixed.

**Comment by rh2048** [comment]
@cshum No dice with using 2 cores on 1.3.7. Still getting the same error after 24h:
<img width="710" alt="image" src="https://user-images.githubusercontent.com/96033673/214185593-63e54328-cb9d-4cc5-b2a1-5beca54b2ae7.png">

**Comment by cshum** [comment]
Follow up from the email, do let me know the result after removing  removing config `IMAGOR_PROCESS_CONCURRENCY` and `VIPS_CONCURRENCY`. If 408 does not occur continuously anymore, we can nail it down to be a queue specific issue. Otherwise we may have to look for something else.

**Comment by rh2048** [comment]
@cshum 

Here's what our status codes look like now:

<img width="875" alt="image" src="https://user-images.githubusercontent.com/96033673/216709047-b61d3ad4-1a1f-4112-9204-bfcaa6535d3b.png">

The first blue line is when we deployed an e2e health check to automatically restart the service, which is why there's not 
many errors after that. It would restart the service every few hours.

The second is the update removing the concurrency restrictions.

The service doesn't seem to completely freeze anymore and our e2e health check hasn't caught any errors since.

*However*, I started getting infrequent 408 errors again yesterday (for now apparent reason), but nowhere near what it was before.

<img width="862" alt="Screenshot 2023-02-03 at 12 59 16" src="https://user-images.githubusercontent.com/96033673/216709671-7c63ad3f-395b-42fb-b9d4-8ba22ff26472.png">

This could be a memory or CPU constraint on the Imagor node, but it seems that every request takes < 2.5s or it completely times out. (See the red arrow above.) I'd imagine that requests would slow down overall if this were the case.

This could also be an issue with our S3 provider. We're using Backblaze on the backend. It's hard to say for certain.

Is there a way to get more detailed logs from Imagor when things time out?

**Comment by cshum** [comment]
Hi @rh2048 you may turn on debug for more verbose logging:
```
DEBUG=1
```
Would you also try these config:
```
IMAGOR_PROCESS_CONCURRENCY=4
IMAGOR_PROCESS_QUEUE_SIZE=8
```
The `IMAGOR_PROCESS_QUEUE_SIZE`, which response with 429 and reject the HTTP request, instead of queuing it up which caused the cascading effect.

**Comment by dangkaka** [comment]
hello, any updates on this issue?
any suggestion for the configuration below?
- `VIPS_CONCURRENCY `
- `IMAGOR_PROCESS_CONCURRENCY`
- `IMAGOR_PROCESS_QUEUE_SIZE`

thanks a lot

**Comment by cshum** [comment]
Please test the docker build at master branch, with the PR @sonda2208:
https://github.com/cshum/imagor/pkgs/container/imagor/89834697?tag=master

```
ghcr.io/cshum/imagor@sha256:6ebb0c613cc09ebebf525fea72bc20b8bff5b9ec8b326528db0b0adc46836807
```

**Comment by cshum** [comment]
Available at v1.4.4
